packages/block-editor/src/components/block-preview/index.js

```diff
@@ -28,12 +32,17 @@ function BlockPreview( props ) {
 	);
 }
 
-export function BlockPreviewContent( { name, attributes, innerBlocks, settings } ) {
-	const block = createBlock( name, attributes, innerBlocks );
+export function BlockPreviewContent( { blocks, settings } ) {
+	if ( ! blocks ) {
+		return null;
+	}
+
+	const theBlocks = castArray( blocks ).map( ( { name, attributes, innerBlocks } ) => createBlock( name, attributes, innerBlocks ) );
```

Comment:
Still not sure why we dod this and not pass the blocks directly?

Reply:
Yeh again, when I don't do this `createBlock` the editor crashes. This code is as per the `master` branch only I'm mapping over a collection of Blocks rather than a single Block.

I know that Reusable Blocks passes `block.initialAttributes` as `attributes` rather than `block.attributes` and so it could be that this is why we need to create new Blocks?

Needs testing to see what happens if we remove this line and just pass Blocks directly.

Reply:
The issue here is we forgot to cast the `value` property, so when [the flattenBlocks() function tries to destruct it](https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/store/reducer.js#L90) and rebuild it as the new `stack` array it triggers the Uncaught TypeError: object is not iterable (cannot read property error). Pretty easy to reproduce it:

```js
var obj = { a: 0, b: 1, c: 2, d: 3, e: 4 };
var arr = [ ...obj ];
```

Adding the commit to cast the `value` avoiding the creating blocks flow.

Reply:
> I know that Reusable Blocks passes block.initialAttributes as attributes rather than block.attributes and so it could be that this is why we need to create new Blocks?

This is what pointed me into the right direction. Blocks only have `attributes`, never `initialAttributes`. We've been using a wrong object with `cloneBlock`! Longer explanation https://github.com/WordPress/gutenberg/pull/16033#discussion_r308998056